### PR TITLE
[Merged by Bors] - feat(GroupTheory/Exponent): lemmas around Monoid.exponent and Subgroup/Submonoid

### PR DESCRIPTION
**Pull request by adri326**
Proves some helpful properties of `Monoid.exponent` on `Submonoid G` and `Subgroup G`:
- the exponent of a group is equal to 1 iff the monoid is trivial (previously only the reverse implication was proven)
- the exponent of `G` divides the exponent of `H` if there exists a multiplication-preserving injection from `G` to `H`
- `Monoid.exponent G = Monoid.exponent H` if `G ≃* H`
- `Monoid.exponent ⊤ = Monoid.exponent G`
- `g ^ Monoid.exponent H = 1` if `g ∈ H`
- generalizes `one_lt_exponent` to `LeftCancelMonoid`

---

This PR belongs to my series of PRs around the formalization of [Rubin's theorem](https://arxiv.org/abs/2203.05930), although these lemmas are only really needed for one step of the proof.

A small tweak will be necessary once #9975 merges.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/GroupTheory/Exponent.lean [hunk 1]
```diff
@@ -356,6 +412,19 @@ theorem exponent_ne_zero_of_finite [Finite G] : exponent G ≠ 0 :=
...
+    1 < Monoid.exponent G := by
+  let _inst := Fintype.ofFinite G
+  obtain ⟨g, hg⟩ := exists_ne (1 : G)
+  rw [← Monoid.lcm_orderOf_eq_exponent]
+  have hg' : 2 ≤ orderOf g := Nat.lt_of_le_of_ne (orderOf_pos g) <| by
+    simpa [eq_comm, orderOf_eq_one_iff] using hg
+  refine hg'.trans <| Nat.le_of_dvd ?_ <| Finset.dvd_lcm (by simp)
+  rw [Nat.pos_iff_ne_zero, Ne.def, Finset.lcm_eq_zero_iff]
+  rintro ⟨x, -, hx⟩
+  exact (orderOf_pos x).ne' hx
```
Comment (tb65536) [inline Mathlib/GroupTheory/Exponent.lean:426]:
```suggestion
  rw [Nat.one_lt_iff_ne_zero_and_ne_one]
  exact ⟨exponent_ne_zero_of_finite, mt exp_eq_one_iff.mp (not_subsingleton G)⟩
```

**Review by Ruben-VandeVelde — APPROVED** [review]
Looks good; one question:

File: Mathlib/GroupTheory/Exponent.lean [hunk 2]
```diff
@@ -335,8 +348,51 @@ theorem lcm_orderOf_eq_exponent [Fintype G] : (Finset.univ : Finset G).lcm order
...
+  exponent_dvd_of_forall_pow_eq_one fun g => e_inj (by
+    rw [map_pow, pow_exponent_eq_one, map_one])
+
+/--
+If there exists a multiplication-preserving equivalence between `G` and `H`,
+then the exponent of `G` is equal to the exponent of `H`.
+-/
+@[to_additive "If there exists a addition-preserving equivalence between `G` and `H`,
+then the exponent of `G` is equal to the exponent of `H`."]
+theorem exponent_eq_of_mulEquiv (e : Nonempty (G ≃* H)) : Monoid.exponent G = Monoid.exponent H :=
```
[the remark this answers is not in the document]
Comment (adri326) [reply Mathlib/GroupTheory/Exponent.lean:370]: You're right, I must have forgotten that equality is a `Prop`, so no data from `e` can be extracted from the theorem's result

**Comment by Ruben-VandeVelde** [comment]
Thanks!

maintainer merge

**Review by Vierkantor — APPROVED** [review]
Thanks!

bors merge

**Pull request closed without merging**